packages/cli/e2e-test/helpers.js

```diff
@@ -117,12 +117,29 @@ async function waitForPageWithText(
     }
   }
 
+  // The page may not be fully loaded and hence we need to retry.
+  const maxAttemptsToSearchText = 3;
+  let attemptsToSearchText = 0;
   const escapedText = text.replace(/"/g, '\\"');
-  browser.assert.evaluate(
-    `Array.from(document.querySelectorAll("*")).some(el => el.textContent === "${escapedText}")`,
-    true,
-    `expected to find text ${text}`,
-  );
+  for (;;) {
+    try {
+      browser.assert.evaluate(
+        `Array.from(document.querySelectorAll("*")).some(el => el.textContent === "${escapedText}")`,
+        true,
+        `expected to find text ${text}`,
+      );
+      break;
+    } catch (error) {
+      if (error instanceof browser.assert.AssertionError) {
+        attemptsToSearchText++;
+        if (attemptsToSearchText <= maxAttemptsToSearchText) {
+          await new Promise(resolve => setTimeout(resolve, intervalMs));
```

Comment:
I'm thinking we might want to do another `visit()` here before we wait

Reply:
Added. Not 100% sure about the reason. But again, headless browsers are mysterious, so it's better to be conservative. :P